### ISSUE TITLE
Also missing support for TF 1.3.0 optional() with second argument, the attribute default value.

### ISSUE THREAD

**Issue by sarath-s-kumar** [tenable/terrascan#1532]
Also missing support for TF 1.3.0 optional() with second argument, the attribute default value.

```failed to build unified config. errors:
variables.tf:5,27-74: Invalid type specification; Optional attribute modifier expects only one argument: the attribute type.```

_Originally posted by @ambis in https://github.com/tenable/terrascan/issues/1331#issuecomment-1257892023_

**Comment by mloskot** [comment]
https://github.com/tenable/terrascan/issues/1331#issuecomment-1257892023 to fix the cross referencing broken due to terrible formatting above

**Comment by adhodgson1** [comment]
I just wanted to add a quick comment as I didn't think the comments in this thread outlined the issue clearly.
Using Terrascan v1.18.1
In my setup the optional value in the strongly typed map works:
```
variable "single_databases_configuration" {
  description = "List of databases configurations"
  type = list(object({
    name                        = string
    sku_name                    = optional(string)
[...]
  }))
}
```
However what I am seeing and I believe is the intent of this issue is that when we add the attribute default value to the variable definition Terrascan throws an error:
```
variable "single_databases_configuration" {
  description = "List of databases configurations"
  type = list(object({
    name                        = string
    sku_name                    = optional(string, "Standard")
[...]
  }))
}
```
This is valid syntax from Terraform 1.3, however Terrascan complains:

Scan Errors -

        -----------------------------------------------------------------------

        IaC Type            :   terraform
        Directory           :   [...]
        Error Message       :   diagnostic errors while loading terraform config dir '[...]'. error from terraform:
[...]\variables.tf:44,32-45: Invalid type specification; Optional attribute modifier expects only one argument: the attribute type.

I use Terrascan as part of Github Super-Linter and due to the error parsing the Terraform code using this feature I am having to disable it completely from the run (i.e, I don't think I can fix this with an ignore rule).

**Comment by mloskot** [comment]
FYA, as follow-up to this issue, I've asked about Terrascan status here
- https://github.com/tenable/terrascan/discussions/1569

**Comment by manish-shanker-osttra** [comment]
Is there any update on this?

**Comment by ricardbejarano** [comment]
Just came accross this. This is why this fails:

1. If you `terrascan scan` with `--log-level debug` it shows:
```
commons/load-dir.go:157 [...] diagnostic errors while loading terraform config dir [...]
```

2. [`load-dir.go:157`](https://github.com/tenable/terrascan/blob/master/pkg/iac-providers/terraform/commons/load-dir.go#L157) executes because `t.parser.LoadConfigDir(dir)` returns more than zero errors in [line 144](https://github.com/tenable/terrascan/blob/master/pkg/iac-providers/terraform/commons/load-dir.go#L144).

3. `t` is a `TerraformDirectoryLoader`, defined in [lines 75-84](https://github.com/tenable/terrascan/blob/master/pkg/iac-providers/terraform/commons/load-dir.go#L75-L84), and its `parser` attribute is of type `*hclConfigs.Parser`.

4. `hclConfigs`, defined in [line 31](https://github.com/tenable/terrascan/blob/master/pkg/iac-providers/terraform/commons/load-dir.go#L31), is the `github.com/hashicorp/terraform/configs` module.

5. If you look for `github.com/hashicorp/terraform/configs` today, you won't find it, because it **no longer exists**.
It appears to have moved to [`github.com/hashicorp/terraform/internal/configs`](https://github.com/hashicorp/terraform/tree/main/internal/configs).

7. If you pull up the (cached) [documentation](https://pkg.go.dev/github.com/hashicorp/terraform/configs?tab=versions) in pkg.go.dev for `github.com/hashicorp/terraform/configs`, you'll see its latest version was published in May 6th, 2021.
I assume this was the last one to ever exist, and thus the one you pull from the [proxy.golang.org](https://sum.golang.org/) during build time.

9. From Terraform's [CHANGELOG](https://github.com/hashicorp/terraform/blob/v1.3/CHANGELOG.md) we know `module_variable_optional_attrs` stopped being an experiment with the release of v1.3, which was released in Sep 21st, 2022, over one year later.

I would assume the fix is to point `terrascan` to the new `github.com/hashicorp/terraform/configs` package, but I don't have time for that today.

cc #1580, #1468, #1028

**Comment by ricardbejarano** [comment]
Yeah, now that I take another look at that... these packages are now `internal`, you can no longer use them.
This is now too deep of a bikeshed for me, when all I get in return are duplicate tfsec and checkov violations.

**If this isn't fixed it will be Terrascan's death.**

**Comment by yermulnik** [comment]
Yeah, we had used to rely upon Terrascan as part of `super-lunter` bundle, though had to disable it globally as it brought more toil because of this issue rather than benefit 😞 
Looking forward to get this issues fixed 🤞🏻

**Comment by gsatur** [comment]
Yeah, I wasted time on this. I use optional attributes quite a bit in my variables and Terrascan's causing this to fail my pre-commit hooks and pipelines. Moving over to Checkov or Trivy.
